### PR TITLE
Put tgi as common module

### PR DESCRIPTION
**Pull request by yongfengdu**
Put tgi helm charts to common to share between different workloads.

## Description

Reorg the directory to put shared helm charts in common directory.
New CodeGen and ChatQnA will depends on this chart. 

## Issues

n/a

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -3,10 +3,10 @@ CHARTS_DIR := ./helm-charts
 .PHONY: test
 
 test:
-	@for chart in $$(find $(CHARTS_DIR) -mindepth 1 -maxdepth 1 -type d); do \
-		echo "Testing chart: $$chart"; \
-		helm lint $$chart; \
+	@for chart in $$(find $(CHARTS_DIR) -mindepth 1 -maxdepth 2 -type f -name "Chart.yaml"); do \
+		  echo "Testing chart: $$(dirname $$chart)"; \
+		  helm lint $$(dirname $$chart); \
 	done
```
Comment (daisy-ycguo) [inline Makefile:9]: Thank you.

File: helm-charts/common/llm-uservice/README.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# llm-uservice
+
+Helm chart for deploying llm microservice.
+
```
Comment (daisy-ycguo) [inline helm-charts/common/llm-uservice/README.md:4]: better to use bigger case LLM instead, https://en.wikipedia.org/wiki/Large_language_model

File: helm-charts/common/llm-uservice/README.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+llm-uservice depends on tgi, refer to tgi for more config details.
```
Comment (daisy-ycguo) [inline helm-charts/common/llm-uservice/README.md:5]: same as LLM, better to use bigger case TGI

File: helm-charts/common/llm-uservice/charts/tgi/values.yaml [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+replicaCount: 1
+
+hftgi:
+  modelId: bigscience/bloom-560m
+  # modelId: /data/OpenCodeInterpreter-DS-6.7B
+  port: 80
+  volume: /mnt
+
+image:
+  repository: ghcr.io/huggingface/text-generation-inference
```
Comment (daisy-ycguo) [inline helm-charts/common/llm-uservice/charts/tgi/values.yaml:17]: I think we have specific opea image for TGI. Will you confirm?

**Review by daisy-ycguo — CHANGES_REQUESTED** [review]

File: helm-charts/common/llm-uservice/templates/deployment.yaml [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Release.Name }}
+          env:
+            - name: TGI_LLM_ENDPOINT
+              value: "http://{{ .Release.Name }}-tgi"
```
Comment (lianhao) [inline helm-charts/common/llm-uservice/templates/deployment.yaml:34]: we should add user configurable tgi servier port here in the TGI_LLM_ENDPOINT

File: helm-charts/common/llm-uservice/templates/deployment.yaml [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          ports:
+            - name: llm-uservice
+              containerPort: 9000
+              protocol: TCP
+          startupProbe:
+            exec:
+              command:
+              - curl
+              - http://{{ .Release.Name }}-tgi
```
Comment (lianhao) [inline helm-charts/common/llm-uservice/templates/deployment.yaml:49]: ditto

**Review by daisy-ycguo — APPROVED** [review]

File: helm-charts/common/llm-uservice/values.yaml [hunk 7]
```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+# Default values for llm-uservice.
+# This is a YAML-formatted file.
+# Declare variables to be passed into your templates.
+
+replicaCount: 1
+hfToken: "insert-your-huggingface-token-here"
```
Comment (leslieluyu) [inline helm-charts/common/llm-uservice/values.yaml:9]: please align with the environment variable name in README.md , like " HUGGINGFACEHUB_API_TOKEN"

**Review by jfding — APPROVED** [review]
Good to merge and continue.

**Pull request merged**